### PR TITLE
postgresql_query: add query_all_results ret value

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
Fixes  https://github.com/ansible-collections/community.general/issues/838

Adds `query_list` and `query_all_results` ret values

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
postgresql_query

### CODE REVIEW

**Comment by Andersson007** [comment]
ready_for_review

**Comment by Andersson007** [comment]
cc @Dorn- @amenonsen @andytom @archf @jbscalia @kostiantyn-nemchenko @matburt @nerzhul @sebasmannem @tcraxs @wrouesnel @ilicmilan

**Comment by Andersson007** [comment]
ready_for_review

File: plugins/modules/database/postgresql/postgresql_query.py [hunk 1]
```diff
@@ -175,23 +175,45 @@
 
 RETURN = r'''
 query:
-    description: Query that was tried to be executed.
+    description:
+    - Executed query.
+    - When reading several queries from a file, it contains the last one.
```
Comment (tcraxs) [inline plugins/modules/database/postgresql/postgresql_query.py:180]:
```suggestion
    - When reading several queries from a file, it contains only the last one.
```

File: plugins/modules/database/postgresql/postgresql_query.py [hunk 2]
```diff
@@ -38,8 +38,8 @@
     type: dict
   path_to_script:
     description:
-    - Path to SQL script on the remote host.
-    - Returns result of the last query in the script.
+    - Path to an SQL script on the target machine.
```
Comment (tcraxs) [inline plugins/modules/database/postgresql/postgresql_query.py:41]:
```suggestion
    - Path to a SQL script on the target machine.
```

File: plugins/modules/database/postgresql/postgresql_query.py [hunk 3]
```diff
@@ -175,23 +175,45 @@
...
+    description:
+    - List containing results of all queries executed (one sublist for every query).
+      Useful when reading several queries from a file.
+    returned: always
+    type: list
+    elements: list
+    sample: [[{"Column": "Value1"},{"Column": "Value2"}], [{"Column": "Value1"},{"Column": "Value2"}]]
 rowcount:
-    description: Number of affected rows.
+    description: Total number of produced or affected rows.
```
Comment (tcraxs) [inline plugins/modules/database/postgresql/postgresql_query.py:216]: When using a script with multiple queries, it contains an all affected rows. But this is not clear from the description. Please describe it more precise.

**Review by tcraxs — COMMENTED** [review]
only a few hints on the documentation 😉

File: plugins/modules/database/postgresql/postgresql_query.py (hunk 3 above)
Comment (Andersson007) [reply plugins/modules/database/postgresql/postgresql_query.py]: Agreed, needs clarification, thank! fixed

**Comment by Andersson007** [comment]
ready_for_review

**Review by tcraxs — APPROVED** [review]
LGTM

**Comment by Andersson007** [comment]
@tcraxs thanks for reviewing!

**Pull request merged**